app/controllers/solidus_bolt/webhooks_controller.rb

```diff
@@ -20,11 +21,12 @@ def update
     private
 
     def verify_bolt_request
-      hash_header = request.headers['X-Bolt-Hmac-Sha256']
+      hmac_header = request.headers['X-Bolt-Hmac-Sha256']
       signing_secret = SolidusBolt::BoltConfiguration.fetch&.signing_secret || ''
-      bolt_request = hash_header == Base64.encode64(OpenSSL::HMAC.digest("SHA256", params.to_json, signing_secret))
+      payload = params.permit!.to_h[:webhook].to_json
```

Comment:
Why you have used the permit! method? Permitted attributes should have used only when the params is provided to an activerecord method that updates the record. I'm worried the executing permit! On this params will make the whole object permitted, even if it shouldn't be permitted under the handler service. WDYT?

Reply:
I think that params[:webhook] should be enough